### ISSUE TITLE
v2.4 occasionally hangs

### ISSUE THREAD

**Issue by NebelNidas** [shedaniel/SmoothScrollingEverywhere#10]
**Steps to reproduce:**
1. Install SSE v2.4 in a MC 1.15.x Fabric profile.
2. Start MC and go into your keybindings settings (or any other GUI which is scrollable, e.g. the modmenu one, if enough mods are installed).
3. Notice the 2-5 seconds hang. It only occurs the first time you open such a scrollable UI though, after that it works fine until you restart the game again. It has a ~50% chance of happening.

**How it should work:**
![2020-03-31-17-42-41](https://user-images.githubusercontent.com/48808497/78047215-640c0300-7378-11ea-830e-4d3fc457270b.gif)

**What happens 50% of the time:**
![2020-03-31-17-43-28](https://user-images.githubusercontent.com/48808497/78047261-6cfcd480-7378-11ea-9053-7425f1a3ed41.gif)

**Notes:**
It's exactly the same as described in my previous issue, #9. Only that this time it's not related to another mod and it only happens around 50% of the time.

**Comment by shedaniel** [comment]
Can you try with only latest Cloth Config (without SSE installed), the biggest change in SSE 2.4 is Cloth Config.

**Comment by shedaniel** [comment]
Can you also share the logs?

**Comment by NebelNidas** [comment]
Log with SSE v2.4 (not much to see, I suppose):
```
[19:35:59] [main/INFO]: Loading for game Minecraft 1.15.2
[19:36:06] [main/INFO]: [FabricLoader] Loading 5 mods: minecraft@1.15.2, smoothscrollingeverywhere@2.4, fabricloader@0.7.8+build.186, cloth-config2@2.10, me_shedaniel_cloth_basic-math@0.3.0
[19:36:06] [main/INFO]: SpongePowered MIXIN Subsystem Version=0.8 Source=file:/C:/Users/[censored]/AppData/Roaming/.minecraft/libraries/net/fabricmc/sponge-mixin/0.8+build.18/sponge-mixin-0.8+build.18.jar Service=Knot/Fabric Env=CLIENT
[19:36:07] [main/INFO]: Setting user: [censored]
[19:36:11] [main/INFO]: Backend library: LWJGL version 3.2.2 build 10
[19:36:16] [main/INFO]: Narrator library for x64 successfully loaded
[19:36:16] [main/INFO]: Reloading ResourceManager: Default, Programmer Art
[19:36:20] [main/INFO]: OpenAL initialized.
[19:36:20] [main/INFO]: Sound engine started
[19:36:20] [main/INFO]: Created: 1024x1024x4 minecraft:textures/atlas/blocks.png-atlas
[19:36:21] [main/INFO]: Created: 256x128x4 minecraft:textures/atlas/signs.png-atlas
[19:36:21] [main/INFO]: Created: 1024x512x4 minecraft:textures/atlas/banner_patterns.png-atlas
[19:36:21] [main/INFO]: Created: 1024x512x4 minecraft:textures/atlas/shield_patterns.png-atlas
[19:36:21] [main/INFO]: Created: 256x256x4 minecraft:textures/atlas/chest.png-atlas
[19:36:21] [main/INFO]: Created: 512x256x4 minecraft:textures/atlas/beds.png-atlas
[19:36:21] [main/INFO]: Created: 512x256x4 minecraft:textures/atlas/shulker_boxes.png-atlas
[19:36:21] [main/INFO]: Created: 256x256x0 minecraft:textures/atlas/particles.png-atlas
[19:36:22] [main/INFO]: Created: 256x256x0 minecraft:textures/atlas/paintings.png-atlas
[19:36:22] [main/INFO]: Created: 256x128x0 minecraft:textures/atlas/mob_effects.png-atlas
[19:38:15] [main/INFO]: Stopping!
```

**Comment by shedaniel** [comment]
Yeah not much too see, how about only latest Cloth Config (without SSE installed)

**Comment by NebelNidas** [comment]
With only Cloth Config (latest, v2.11.2) installed, everything works as intended. I even installed v2.10, which seems to be the version SSE v2.4 is using, but even that works fine, too. So it must be SSE which has the error. Have you been able to reproduce the issue?

**Comment by NebelNidas** [comment]
Interesting... I just bundled SSE v2.3, which on itself works fine, together with CC v2.11.2. Then it also hanged! So it seems to be Cloth Config, indeed.

**Comment by shedaniel** [comment]
I have tried myself and I've failed to reproduce the issue in v2.4 with CC v2.10

**Comment by shedaniel** [comment]
I will try v2.4 with CC v2.11.2

**Comment by shedaniel** [comment]
I've failed to reproduce the issue in SSE v2.3 with CC v2.11.2, will now try v2.3

**Comment by shedaniel** [comment]
I've failed to reproduce the issue in SSE v2.3 with CC v2.11.2.

**Comment by NebelNidas** [comment]
I've been able to track down the problematic Cloth Config update. It is v2.8.0 which introduced the bug, v2.7.1 is the last without the issue. I can post my computer specs, if that helps.

**Comment by shedaniel** [comment]
I've look into what changes Cloth Config 2.8 has done, and all it does is changed some internal classes used for basic geometry.

It might be useful if you share your PC specs. The impact of v2.8 is minimal, though one thing changed is that Cloth Config started to JIJ, I wonder if JIJ is the issue here.

**Comment by NebelNidas** [comment]
**Hardware:**
- CPU: Intel(R) Core(TM) i7-4790 CPU @ 3.60GHz   3.59 GHz
- GPU: Nvidia Quadro K620 (I know, it's awful, I'll soon get a better one xD)
- RAM: 1600MHz DDR3

**Software:**
- Windows 10 1909 Build 18363.752 x64
- Java 8 Update 241Build 1.8.0_241-b07
- Minecraft Legacy Launcher 1.6.93
- Fabric Loader 0.7.8 Build 186 for 1.15.2

**Comment by shedaniel** [comment]
Nothing seems to out of place, unfortunately I don't really know what's going wrong, as CC 2.8 is a small update. Will continue looking into this bug tomorrow hopefully.
